Adds configuration for new analytics tasks

pomegranited opened the pull request:
Updated hks configuration to deploy updated Insights, Analytics API, and Analytics Pipeline, including new analytics tasks still under review: ModuleEngagementWorkflow, and ProblemResponseReportsWorkflow.

Because hks-epod have not yet upgraded their edxapp instance to eucalyptus, I chose to rebase from `releases/dogwood.rc`, and then selectively cherry-pick the changes applied to edx:master for the analytics-related roles.

From [4d82736](https://github.com/hks-epod/configuration/pull/4/commits/4d827365fc0489020809d086ef08a2063e6189f1)...[26c484d](https://github.com/hks-epod/configuration/pull/4/commits/26c484d77f1d1bd257fe16011854334ca11c5c07):
- rebased from `releases/dogwood.rc`

From [aaa82d7](https://github.com/hks-epod/configuration/pull/4/commits/aaa82d742afb248754c461119f6dda5e028fd9eb)...[d816b4b](https://github.com/hks-epod/configuration/pull/4/commits/d816b4b9987d32cbe562ae0e919fde7a7be0fc37):
- cherry-picked changes to [`playbooks/roles/analytics_api`](https://github.com/edx/configuration/commits/master/playbooks/roles/analytics_api):
  
  ```
  git cherry-pick -x b209e902d24b5bb81d778cc94f8c26c94792c1a8
  git cherry-pick -x 5c50abedc791a1b156d69f0418332d0c2c33e239
  git cherry-pick -x 4ba0ea0cbd8b0c20bc542db7e0f965dc3433bdff
  git cherry-pick -x 1389a660c8a8e93b894be6605d3383d9fd4de9a8
  git cherry-pick -x f899f4fd479441ca793c3f6ff3b084ad0a2916a7
  ```
- cherry-picked changes to [`playbooks/roles/insights`](https://github.com/edx/configuration/commits/master/playbooks/roles/insights):
  
  ```
  git cherry-pick -x a397c58628d922e5bb6b504cb795f5c49ffba110
  git cherry-pick -x 6eb3a9f972dfb60b3b565556a83c1f6797536b59
  git cherry-pick -x b9ab168260df42e76e6fdc28860b5a966dff9f38
  git cherry-pick -x 12d91d7c1a50030a38a2bc6c8d8aedef48cc34bc
  git cherry-pick -x 533400cce7a900676d9c933750d6d83832ec7ad5
  ```
- cherry-picked changes to [`playbooks/roles/analytics-pipeline`](https://github.com/edx/configuration/commits/master/playbooks/roles/analytics_pipeline):
  
  ```
  git cherry-pick -x 1780beaa05ee2a79af4068fdf09cd7e55db28159
  git cherry-pick -x a4457c91eb3fb37573091688e86dbe79da6e3c7b
  git cherry-pick -x ae9e7380e1bd48cfecdef10e46ebd0f6fb98b5ee
  git cherry-pick -x 0f45b4dabed2c93a2906b633056a3b1a39d371c5
  ```

[0237958](https://github.com/hks-epod/configuration/pull/4/commits/02379580f476efea79b0c370227b2d8711d84e55), [a75f89f](https://github.com/hks-epod/configuration/pull/4/commits/a75f89fe865920a33879d91841b9531acb028213), [9da3ba5](https://github.com/hks-epod/configuration/pull/4/commits/9da3ba51cf95f92d86edfd6281b3c3112e1f30e0):
- cherry-picked configuration changes for new analytics tasks:
  
  ```
  git cherry-pick -x 4f3e35dc6ef251099793c9e3491e5dff68ec19d5
  git cherry-pick -x 09aefceacf43e6928539498229eebac891697d47
  git cherry-pick -x 4b4efdfa59c26a32c0792efcff4d47abca136546
  ```

[d4a14ab](https://github.com/hks-epod/configuration/pull/4/commits/d4a14ab1e7569242ca170176b60f6e3c7ec8caad), [bebb956](https://github.com/hks-epod/configuration/pull/4/commits/bebb9563600f2f3c023ef315c29995fa923613c8) fix merge issues noticed during PR review:
- cherry-picked fix
  
  ```
  git cherry-pick -x 713b5cccab5d093050121ffe03de589ceadc157c
  ```

Author notes and concerns:
1. This change shouldn't be merged until the PRs listed in [OC-1786](https://tasks.opencraft.com/browse/OC-1786) are merged into the hks-epod forks.
2. Changes have not been tested with a new edxapp deployment.

feanil committed:
Print only the 10 slowest tasks instead of all of them.

Commit:
Write to stdout so that vagrant doesn't color it red

itsjeyd committed:
Automate preparatory steps for running pipeline acceptance tests.

Companion PR: https://github.com/edx/edx-analytics-pipeline/pull/204

(cherry picked from commit 1780beaa05ee2a79af4068fdf09cd7e55db28159)

mtyaka committed:
Add aws role to edx_sandbox.yml.

mtyaka committed:
New playbook to install analytics services.

mtyaka committed:
Add security role to analytics_sandbox.yml.

jbzdak committed:
Adds provisioning of a Jenkins instance dedicated for analytics tasks.

This jenkins instance will perform following tasks:

* Set up authentication using Jenkins Unix Auth Realm
* Set up administrative user
* Load credentials from a JENKINS_ANALYTICS_CREDENTIALS variable
* Checkout seed job repository
* Launch seed job to populate Jenkins with analytics Jobs

(cherry picked from commit 428cd59bae5173a740aff208a68d92d2ed5d7add)

Conflicts:
	.gitignore
	playbooks/roles/jenkins_master/tasks/main.yml

smarnach committed:
Install s3cmd via pip rather than from a tarball.

smarnach committed:
Allow to provide S3 IAM credentials for log syncing.

mtyaka committed:
Install libffi package on analytics box.

Commit:
Workaround for debian squeeze deprecation and Chrome.

The older chrome repositories have been taken down; however, we
use a saved copy of the debian package. After installing that package,
ensure it is not referenced by the target machine after Chrome
is installed.

jibsheet committed:
Once this mv runs on the base box, it fails on provision

Just ensure the file is gone.

Commit:
Dogwood.1 targets, devstack uses the 3/9 boxes

Commit:
Add a message in the fullstack Vagrantfile also.

mtyaka committed:
Always run mysql role.

Deployment breaks without it for some reason.

mtyaka committed:
Remove demo & analytics roles.

We don't need them.

e-kolpakov committed:
Fix typo - ANALYTICS_PIPELINE_OUTPUT_DATABASE contains `username` key, not `user`

(cherry picked from commit a4457c91eb3fb37573091688e86dbe79da6e3c7b)

Commit:
Pin IDAs to OPENEDX_RELEASE, and remove edx-notes-api from native install

Commit:
Get rabbit ready before we start workers

Commit:
Provisioning Jenkins for DSL jobs PR fixes

(cherry picked from commit ba1bd1fcf09c6d8bac40a51d4fd7e56a1ec0b932)

Commit:
Fix a bug in the git module: branches might not check out properly

If a git version is specified as a branch, and the commit referred to is
already on the local working tree, but the branch is not, then the
checkout will fail.

This is because the Ansible 1.9.3 module would check to see if the
commit was already present, and if so, did not fetch. This meant the
branch wasn't defined in the local working tree.

This can happen if you make a branch but there aren't any changes on the
branch.

(cherry picked from commit ae9e7380e1bd48cfecdef10e46ebd0f6fb98b5ee)

Commit:
Fix a bug in the git module: branches might not check out properly

Cherry-picked from master.

If a git version is specified as a branch, and the commit referred to is
already on the local working tree, but the branch is not, then the
checkout will fail.

This is because the Ansible 1.9.3 module would check to see if the
commit was already present, and if so, did not fetch. This meant the
branch wasn't defined in the local working tree.

This can happen if you make a branch but there aren't any changes on the
branch.

pomegranited committed:
Create Jenkins jobs for analytics tasks using a job DSL repo.

* defaults/main.yml: adds analytics schedule variables for configuring the AnalyticsSeedJob, with all analytics tasks enabled by default.
* templates/seed_job_template.xml: use updated analytics seed job config and description
* tasks/main.yml: generates a hashed password for the jenkins user, and adds jenkins-auth and create-seed-job tags.
* README.md: documents seed job parameters and jenkins auth change

(cherry picked from commit 1cfe1ac06b336a7f07e00c63ac4b991331ecc40d)

Commit:
Remove all mention of devpi, which we no longer use

Seems like we stopped using devpi with #971 two years ago.

Cherry-picked from commit 11d696b.

Commit:
Update wiki links in file headers

The headers were pointing to the old github wiki. Since we've officially
moved to Confluence, these links should be updated.

(cherry picked from commit b209e902d24b5bb81d778cc94f8c26c94792c1a8)

Conflicts:
	playbooks/roles/asqatasun/defaults/main.yml
	playbooks/roles/asqatasun/tasks/main.yml
	playbooks/roles/container/defaults/main.yml
	playbooks/roles/container/meta/main.yml
	playbooks/roles/container/tasks/main.yml
	playbooks/roles/credentials/defaults/main.yml
	playbooks/roles/credentials/tasks/main.yml
	playbooks/roles/devpi/handlers/main.yml
	playbooks/roles/devpi/tasks/main.yml
	playbooks/roles/discovery/defaults/main.yml
	playbooks/roles/discovery/meta/main.yml
	playbooks/roles/discovery/tasks/main.yml
	playbooks/roles/docker/defaults/main.yml
	playbooks/roles/docker/meta/main.yml
	playbooks/roles/docker/tasks/main.yml
	playbooks/roles/enhanced_networking/tasks/main.yml
	playbooks/roles/go-agent-docker-server/defaults/main.yml
	playbooks/roles/go-agent-docker-server/tasks/main.yml
	playbooks/roles/go-agent/defaults/main.yml
	playbooks/roles/go-agent/meta/main.yml
	playbooks/roles/go-agent/tasks/main.yml
	playbooks/roles/go-server/defaults/main.yml
	playbooks/roles/go-server/meta/main.yml
	playbooks/roles/go-server/tasks/main.yml
	playbooks/roles/mount_ebs/tasks/main.yml

Commit:
Add show-repo-heads script for helping with diagnosis

Commit:
Null API keys for Analytics AWS

(cherry picked from commit 5c50abedc791a1b156d69f0418332d0c2c33e239)

Conflicts:
	playbooks/roles/analytics_api/defaults/main.yml

Commit:
Make jenkins_analytics install its required dependency

(cherry picked from commit 9c0f307729ddfc28defe636007700df2491e7229)

Commit:
Add usage example

(cherry picked from commit b97ca25fa183ba41defc5bf6020e132417ee53a4)

pomegranited committed:
Various cleanup fixes for analytics jenkins tasks

* Uses JENKINS_HOME=/edx/var/jenkins.  This addresses the pyinstrument
  re-install issue seen when JENKINS_HOME is a symlink.
* Installs libffi-dev and libpq-dev packages with analytics_jenkins playbook
* Adds an SSH Credential, which is passed as the MASTER_SSH_CREDENTIAL_ID
  parameter to the seed job.
* When the seed job is run, ignore unreferenced jobs, instead of deleting them.
  This allows you to run and re-run the seed job from Jenkins to update a
  particular analytics job, and untick the jobs you don't want to update.
* Lets <ANALYTICS_TASK>_EXTRA_VARS be a @path/to/file.yml

(cherry picked from commit 1dd291abc67cbf1c4e92485419bb882c9090d165)

pomegranited committed:
Uses Github OAuth plugin to secure Jenkins

* Configures github oauth as default jenkins auth realm (was unix system user)
* Uses the Matrix Authorization Strategy at 3 levels: anonymous, administrator,
  and job builder.
* Allows configuration of the list of users/groups and permissions granted to
  each authorization level.
* Adds ssh private key to jenkins user, to maintain CLI tool access
* Sets jenkins password only if using old unix system user realm; locks the
  jenkins system user when using github oauth.
* Adds jenkins public key to Jenkins user config.xml, and creates that user
  config.xml file if it's missing (required to bootstrap jenkins user CLI access)
* Renames the existing JENKINS_ANALYTICS_GITHUB_* credential variables to
  JENKINS_ANALYTICS_GITHUB_CREDENTIAL_* to avoid confusing them with the Github
  OAUTH variables.

(cherry picked from commit a2caff27e02fc120bde21a37c1df94c30b2a6ebb)

e-kolpakov committed:
Enabling SSL for insights

e-kolpakov committed:
Dual SSL and non-SSL config for insights

pomegranited committed:
Converts the seed job xml to groovy DSL, and incorporates changes from PR #2962

* Builds the seed job xml from groovy DSL using the job-dsl-plugin's gradle run task.
  Ref https://github.com/jenkinsci/job-dsl-plugin/wiki/User-Power-Moves
* Adds COMMON_VARS text parameter
* Uses DSL_BRANCH and SECURE_BRANCH parameter values for seed job git repos
* Removes unnecessary jinja formatting from seedJob.groovy template

(cherry picked from commit 34ef9de3f13b38b0078fd9c2ffcf18d4ef58fa1a)

Commit:
Update Vagrantfiles for dogwood.2

mulby committed:
Remove boto connection default for analytics API elasticsearch config (#2958)
(cherry picked from commit 4ba0ea0cbd8b0c20bc542db7e0f965dc3433bdff)

Commit:
Need to point to library for custom modules

Commit:
Pin things installed by devstack and fullstack

mtyaka committed:
Add aws role to edx_sandbox playbook.

Needed to set up S3 syncing of tracking logs.

e-kolpakov committed:
Configurable jenkins nginx port

Commit:
Update Vagrantfiles for dogwood.3

Commit:
Source nodeenv before running npm

the extra e is for extra

(cherry picked from commit 6eb3a9f972dfb60b3b565556a83c1f6797536b59)

Conflicts:
	playbooks/roles/insights/tasks/main.yml

pomegranited committed:
Adds jenkins_protocol_https

When set to 'true' (default value), configures nginx with proxy_redirect from http: to https:

When set to 'false', the proxy allows http: requests without redirecting to https:.

Ref https://github.com/edx/edx-analytics-configuration/blob/master/roles/jenkins/templates/etc/nginx/sites-available/jenkins.j2#L8-L11

(cherry picked from commit a46bb8cb484348233cc34e60ddade6b9bd02ca06)

pomegranited committed:
Uses proxy_set_header Host $http_host

$http_host is the same as $host, except it also includes the port number if
present in the request.

This is required for valid proxied content rewriting if jenkins_nginx_port is
set to something other than the default port 80.

(cherry picked from commit 2b977ac0f742ea2965cfb30f1c2ee64129463d0b)

pomegranited committed:
Allows JENKINS_ANALYTICS_AUTH_REALM: none

Useful for Jenkins installations which are secured via other means, e.g. firewall rules.

* Adds an authentication realm option which disables Jenkins authentication.
* Configures CLI authentication for existing auth realms
* Fixes a typo in execute_jenkins_cli.yml "Wait for Jenkins CLI"

(cherry picked from commit 06aa5a9ca41ebaa367ebb9f731a37b44d9e3709a)

jibsheet committed:
Clean up /edx/var/edxapp/staticfiles before paver runs

When running on a dirty image (such as a sandbox base image or an AMI
build) this clears static files so that collectstatic will copy
everything.

Django's collectstatic copies files and doesn't preserve timestamps,
then only copies "newer modified files".  This leads to a condition
where old static files are in /edx/var/edxapp/staticfiles with a
timestamp of Yesterday, and we add a new dep which comes down from npm
with a timestamp of last week.  Collectstatic will not copy those.

(cherry picked from commit 713b5cccab5d093050121ffe03de589ceadc157c)

Commit:
convert sudo_user to become user in roles for Ansible 2.x

(cherry picked from commit 1389a660c8a8e93b894be6605d3383d9fd4de9a8)

Conflicts:
	playbooks/roles/aws/tasks/main.yml

Commit:
Defining logout_uri for OAuth clients

ECOM-4678

(cherry picked from commit 533400cce7a900676d9c933750d6d83832ec7ad5)

Conflicts:
	playbooks/roles/credentials/defaults/main.yml
	playbooks/roles/discovery/defaults/main.yml

Commit:
parameters written as fully expanded YAML dictionaries

(cherry picked from commit b9ab168260df42e76e6fdc28860b5a966dff9f38)

Conflicts:
	playbooks/roles/git_clone/tasks/main.yml
	playbooks/roles/go-server/tasks/download_backup.yml
	playbooks/roles/go-server/tasks/main.yml
	playbooks/roles/go-server/tasks/restore_backup.yml
	playbooks/roles/go-server/tasks/setup_regular_backup.yml
	playbooks/roles/harstorage/tasks/main.yml

clintonb committed:
Added OpenID Connect logout URL to all IDAs and the IDA template (#3133)

ECOM-4609
(cherry picked from commit 12d91d7c1a50030a38a2bc6c8d8aedef48cc34bc)

Conflicts:
	playbooks/roles/credentials/defaults/main.yml
	playbooks/roles/discovery/defaults/main.yml

Commit:
Apply stuck-worker fix from master

Commit:
using YAML syntax for rewriting the tasks

(cherry picked from commit 0f45b4dabed2c93a2906b633056a3b1a39d371c5)

pomegranited committed:
Adds variables to the analytics_api master config YAML file (#3273)

* Adds DATE_FORMAT and DATETIME_FORMAT to the analytics_api config .yml
* Adds DEFAULT_PAGE_SIZE and MAX_PAGE_SIZE variables to the analytics_api master .yml file. See https://github.com/edx/edx-analytics-data-api/pull/128 for how they're used.
* PR review: fixed page size defaults to match app
(cherry picked from commit f899f4fd479441ca793c3f6ff3b084ad0a2916a7)

Conflicts:
	playbooks/roles/analytics_api/defaults/main.yml

Commit:
Adding variables needed for report download

(cherry picked from commit 4b4efdfa59c26a32c0792efcff4d47abca136546)

Conflicts:
	playbooks/roles/analytics_api/defaults/main.yml

Commit:
Adding variables needed for report download

(cherry picked from commit 4f3e35dc6ef251099793c9e3491e5dff68ec19d5)

Commit:
Ecommerce and Programs need to play along with OPENEDX_RELEASE

(cherry picked from commit 327a57328a0f515cab79af6e774e11c3ae52bffc)

Commit:
Ecommerce and programs support

(cherry picked from commit 6ec18d7ae4339a468fbe4eddaa498db4c9dd1b0b)

pomegranited committed:
Adds INSIGHTS_LEARNER_API_LIST_DOWNLOAD_FIELDS

which is used by the Learner List view's "Download CSV" button to
determine the fields to include in the CSV, and their sort order.

(cherry picked from commit 09aefceacf43e6928539498229eebac891697d47)

itsjeyd commented on playbooks/roles/edxapp/tasks/deploy.yml:
```diff
@@ -354,19 +373,19 @@
+    state: started
+  become_user: "{{ supervisor_service_user }}"
   when: celery_worker is not defined and not disable_edx_services
   with_items: service_variants_enabled
```
@pomegranited I don't see `{{ item }}` mentioned above -- is this still going to do the right thing? If it is, could this line be removed?

pomegranited replied:
@itsjeyd Oops, you're absolutely correct.  The merge misplaced that `with_items` statement.  Fixed in [51f0ad0](https://github.com/hks-epod/configuration/commit/51f0ad00a1dff9ee00ce9ae46060219ff0802814).

itsjeyd commented:
@pomegranited As discussed on IRC, I skimmed the commits from the rebase, and had a closer look at the commits you cherry-picked on top of that (focusing on checking conflict resolution). In addition to the in-line comments that I left, here is what I noticed:
- Looks like `set custom plugin permissions` task got removed from `playbooks/roles/jenkins_master/tasks/main.yml` in commit https://github.com/hks-epod/configuration/pull/4/commits/a5338f3ab935ddf758f1cfe9d1eb4160a41a9039 (this is one of the commits that were added as part of the rebase).
- Commit https://github.com/hks-epod/configuration/pull/4/commits/58e5ca8ec337ac0826a68147176120f15cedfe33 seems to be missing changes to files that had conflicts, please double-check (compare to [original commit](https://github.com/hks-epod/configuration/commit/b9ab168260df42e76e6fdc28860b5a966dff9f38)).
- For the same commit, diff the `playbooks/roles/edxapp/tasks/service_variant_config.yml` file against the version from the [original commit](https://github.com/hks-epod/configuration/commit/b9ab168260df42e76e6fdc28860b5a966dff9f38); it looks like some changes are missing in your version vs. the version from the original commit.
- Commit https://github.com/hks-epod/configuration/pull/4/commits/ad6f46c57361875e766a7ee2dccb44f745f6021f seems to be missing changes to files that had conflicts, please double-check (compare to [original commit](https://github.com/hks-epod/configuration/commit/12d91d7c1a50030a38a2bc6c8d8aedef48cc34bc)).
- Commit https://github.com/hks-epod/configuration/pull/4/commits/61b5b186ea979eadcb384c4534e4870631ded0ef seems to be missing changes to files that had conflicts, please double-check (compare to [original commit](https://github.com/hks-epod/configuration/pull/4/commits/61b5b186ea979eadcb384c4534e4870631ded0ef)).

I did not have time to check conflict resolution for https://github.com/hks-epod/configuration/pull/4/commits/26c484d77f1d1bd257fe16011854334ca11c5c07 (one of the commits added as part of the rebase). Since there's quite a number of files in that one that had conflicts, it would be worth double-checking on your end.

pomegranited commented on playbooks/roles/edxapp/tasks/deploy.yml:
```diff
@@ -354,19 +373,19 @@
+    state: started
+  become_user: "{{ supervisor_service_user }}"
   when: celery_worker is not defined and not disable_edx_services
   with_items: service_variants_enabled
```
@itsjeyd I believe the `with_items` statement below creates the `{{items}}`.

The previous code restarted the whole `edxapp:` group several times, once for each `service_variants_enabled`.  But the updated code actually restarts each individual service, which looks like what they were intending in the first place.

itsjeyd replied:
@pomegranited I might be missing something, but in the new version of the code, there's just `name: "edxapp:"` (instead of `name: "edxapp:{{ item }}"`), but the new version still has that `with_items: service_variants_enabled` line. So how can the info that comes from `with_items` be taken into account by the `supervisorctl` module?

I think we'd want to run `supervisorctl` with `edxapp:lms`, `edxapp:cms`, etc. instead of running it with `edxapp:` multiple times for each item that `with_items` generates?

pomegranited committed:
PR review: moves "with_items" statement misplaced by merge.

Merge had shifted "with_items: service_variants_enabled" into the "ensure
edxapp has started" task, but it belongs in "restart edxapp".

pomegranited commented:
@itsjeyd Thank you for your thorough review!  Please see my comments below, and let me know if you think this branch is now ready to merge.
- I'm still seeing the [`set custom plugin permission` task in this branch](https://github.com/hks-epod/configuration/pull/4/files#diff-d887fe1442a4be97ae2d3183ac49489eR144), so I think that's ok.  Perhaps it got re-added by a subsequent changeset?
- For `playbooks/roles/edxapp/tasks/service_variant_config.yml`, I've cherry-picked [713b5cc](https://github.com/hks-epod/configuration/commit/713b5cccab5d093050121ffe03de589ceadc157c) to add in the missing stanza.
- Fixed that misplaced `with_items` statement you found with [51f0ad0](https://github.com/hks-epod/configuration/commit/51f0ad00a1dff9ee00ce9ae46060219ff0802814).  Good find, thank you!
- The missing conflict files from [58e5ca8](https://github.com/hks-epod/configuration/pull/4/commits/05aa3d3e3a8adc6a2ffdbfc405b533473c9753db) and [ad6f46c](https://github.com/hks-epod/configuration/pull/4/commits/8107be8b43b76121d1b742482592f0e799b4b150) are missing because the roles they refer to don't exist yet in this branch.
  I've grepped through this branch and found no reference to these roles.  Are there any in this list that worry you?
  - [roles/git_clone](https://github.com/edx/configuration/commits/master/playbooks/roles/git_clone) - added June 1 2016
  - [roles/go_server](https://github.com/edx/configuration/commits/master/playbooks/roles/go-server) - added 24 Feb 2016
  - [roles/harstorage](https://github.com/edx/configuration/commits/master/playbooks/roles/harstorage)  - added Mar 2 2016
  - [roles/discovery](https://github.com/edx/configuration/commits/master/playbooks/roles/discovery) - added Jan 1 2016
  - [roles/credentials](https://github.com/edx/configuration/commits/master/playbooks/roles/credentials) - added Dec 17 2015
- The diff for [26c484d](https://github.com/hks-epod/configuration/pull/4/commits/d456804b07e01bc395edc5d8bad31f2c3242ff5d) looks fine.  It also contains several conflict files that are missing from this branch, but the [original commit](https://github.com/hks-epod/configuration/commit/b209e902d24b5bb81d778cc94f8c26c94792c1a8) only affects the comments at the top of the files, so I'm not concerned.
- The [edx_sandbox.yml lists the `aws` role twice](https://github.com/hks-epod/configuration/blob/ec80f096708515d1e696fdfacf0ac13ee3ece422/playbooks/edx_sandbox.yml) ([L23](https://github.com/hks-epod/configuration/blame/ec80f096708515d1e696fdfacf0ac13ee3ece422/playbooks/edx_sandbox.yml#L23) and [L33](https://github.com/hks-epod/configuration/blame/ec80f096708515d1e696fdfacf0ac13ee3ece422/playbooks/edx_sandbox.yml#L33)).  Decided to leave it alone, as it won't hurt anything.

@itsjeyd _EDIT_ Oh bugger.. I just rebased this PR, just trying to merge the latest changes from [PR #3305](https://github.com/edx/configuration/pull/3305) into the `INSIGHTS_LEARNER_API_LIST_DOWNLOAD_FIELDS`, and it changed almost every single commit hash!  I'm so sorry.  I've fixed my links above so they refer to the correct location.  I won't do that again.

itsjeyd commented:
@pomegranited Thanks for the updates!

> The missing conflict files from 58e5ca8 and ad6f46c are missing because the roles they refer to don't exist yet in this branch. I've grepped through this branch and found no reference to these roles. Are there any in this list that worry you?

No, not really. `git_clone` and `credentials` seemed "suspicious" at first, but IIUC the former is the result of a refactoring (i.e., the tasks it contains should still be present on this branch, just spread out over other roles). And for the latter, it seems like we would eventually want to move to using the credentials service, but it might not be needed in a Dogwood context?

> The edx_sandbox.yml lists the aws role twice (L23 and L33). Decided to leave it alone, as it won't hurt anything.

I think that's because you cherry-picked both https://github.com/hks-epod/configuration/pull/4/commits/a8c99f180a59487f65a1ee442c129f4d160d259d and https://github.com/hks-epod/configuration/pull/4/commits/88f49c2495a66dcd2bb381e588545f95117aa5da? If that's a result of rebasing (i.e., if both of these commits are in `named-release/dogwood.rc`, I'm fine with keeping both of them. Otherwise you might be able to remove one of the commits?

All in all, this looks like it's good to go. The rebase made it a bit difficult to go back and reference the commits that I originally commented on. But you addressed all of the individual items in your [reply](https://github.com/hks-epod/configuration/pull/4#issuecomment-242386041), and my remaining comments are just questions/nits, so :+1:

pomegranited committed:
Re-adds the jenkins_custom_plugins default variable

This change was somehow mis-merged during the rebase?

Caused the jenkins-analytics.yml playbook to fail during deployment.

pomegranited committed:
Removes second `aws` role added to edx_sandbox.yml

by 88f49c2495a66dcd2bb381e588545f95117aa5da

pomegranited commented:
Thanks, @itsjeyd !

Re: `aws` role duplication:

Yes, those commits are a result of rebasing, but oddly, the `aws` role doesn't appear in [`named-release/dogwood.rc`'s edx_sandbox.yml](https://github.com/edx/configuration/blob/named-release/dogwood.rc/playbooks/edx_sandbox.yml) at all!  So it's probably some kind of merge issue.  But since @mtyaka went to the trouble of adding it twice, I've decided to keep one.

pomegranited committed:
Merge branch 'epodx' into jill/epodx-analytics

Conflicts:
	playbooks/edx_sandbox.yml
